# Algortihm Parameters
alpha: 0.1
gamma: 0.8
epsilon: 0.9
epsilon_discount: 0.999 # 2197 eps to reach 0.1, 587 to reach 0.5
# epsilon_discount: 0.998 # 1097 eps to reach 0.1, 587 to reach 0.5
nepisodes: 120
nsteps: 500

# Environment Parameters
desired_pose:
  x: 0.0
  y: 0.0
  z: 0.09 # 0.13 m
desired_force: 7.08 # In Newtons, normal contact force when stanting still with 9.81 gravity
desired_yaw: 0.0 # Desired yaw in radians for the robot to stay
desired_roll: 0.0 # Desired yaw in radians for the robot to stay
desired_pitch: 0.25 # Desired yaw in radians for the robot to stay
max_height: 1.0   # in meters
min_height: 0.0   # in meters
max_incl: 1.6       # in rads
running_step: 0.0010   # in seconds
joint_increment_value: 0.2   # in radians
done_reward: 1000000 # reward
alive_reward: 100.0 # reward

list_of_observations: # Observations you want to use for the learning
  - distance_from_desired_point
  - base_z
  - base_roll
  - base_pitch
  # - base_yaw
  #- base_angular_vel_x
  #- base_angular_vel_y
  #- base_angular_vel_z
  #- base_linear_acceleration_x
  #- base_linear_acceleration_y
  #- base_linear_acceleration_z
  
  # maybe add
  # - contact_force
  - joint_states
  # - joint_effort
  - touching_ground

discrete_division: 30 # Parts in which we divide each of the observations sensory range

weight_r1: 3.0 # Weight for joint positions ( joints in the zero is perfect )
weight_r2: 1.0 # Weight for joint efforts ( no efforts is perfect )
weight_r3: 0.0 # Weight for contact force similar to desired ( weight of hexapod )
weight_r4: 100.0 # Weight for orientation ( vertical is perfect )
weight_r5: 10000.0 # Weight for distance from desired point ( on the point is perfect )
weight_r6: 20.0 # Weight for not touching floor
weight_r7: 30.0 # Weight for not synchronisation
# Tibia is furthest from the body
init_joint_pose:
  coxa_l1: 0
  coxa_l2: 0
  coxa_l3: 0
  coxa_r1: 0
  coxa_r2: 0
  coxa_r3: 0
  tibia_l1: 0
  tibia_l2: 0
  tibia_l3: 0
  tibia_r1: 0
  tibia_r2: 0
  tibia_r3: 0
  femur_l1: 1
  femur_l2: 1
  femur_l3: 1
  femur_r1: 1
  femur_r2: 1
  femur_r3: 1

joint_limits_array:
  min: -1.5
  max: 1.5

episode_done_criteria:
  # - monoped_minimum_height
  #- monoped_vertical_orientation
  - stand_up
  - less_exceeded_joint_position
maximum_base_linear_acceleration: 3000.0 # Based on making fall from a body height vertical max/min imu readings

maximum_base_angular_velocity: 0.000031998 # Based on free fall, one body length, at 45 degree angle max/min imu readings 20.0

maximum_joint_effort: 0.2745862 
